pkgs/build-support/php/hooks/php-script-utils.bash

```diff
@@ -10,3 +11,26 @@ setComposeRootVersion() {
 
     set -e
 }
+
+checkComposerValidate() {
+    if ! composer validate --strict --no-ansi --no-interaction; then
+        if [ "1" == "${composerStrictValidation-}" ]; then
+            echo
+            echo -e "\e[31mERROR: composer files validation failed\e[0m"
+            echo
+            echo -e '\e[31mThe validation of the composer.json and composer.lock failed.\e[0m'
+            echo -e '\e[31mMake sure that the file composer.lock is consistent with composer.json.\e[0m'
+            echo
+            exit 1
+        else
+            echo
+            echo -e "\e[33mWARNING: composer files validation failed\e[0m"
+            echo
+            echo -e '\e[33mThe validation of the composer.json and composer.lock failed.\e[0m'
+            echo -e '\e[33mMake sure that the file composer.lock is consistent with composer.json.\e[0m'
+            echo
+            echo -e '\e[33mThis check is not blocking, but it is recommended to fix the issue.\e[0m'
+            echo
```

Comment:
IMO a better way would be to run `composer validate`, and if it fails run it again with `--no-check-lock` and if that passes then you know for sure the error was the lock file outdated, and you can output this msg. If it's not that then I'd probably fail hard.

Because there are many other reasons this could fail, and the messages above only suggest it is about the lock file.

Reply:
Thanks!

What do you think about this new PR?

https://github.com/NixOS/nixpkgs/pull/293582